### DISCUSSION TITLE
Set 2 variables from rule

### DISCUSSION THREAD

**Discussion by garyvdm** [arendst/Tasmota#21713] [Q&A] [answered]
I would like to set 2 variables from a rule. This is what I've tried (plus a few other variations):

```
Rule2
  ON time#minute<240 DO Mem1 0; Mem2 1 break
  ON time#minute<360 DO Mem1 35; Mem2 40 break
  ON time#minute<540 DO Mem1 0; Mem2 1 break
  ON time#minute<900 DO Mem1 52; Mem2 60 break
  ON time#minute<1440 DO Mem1 0; Mem2 1 break
```

What I hoped is that say for for `time#minute<900`, `Mem1` would be set to `52`, and `Mem2` would be set to `60`. 

What happens is: 

```
10:41:00.290 RUL: TIME#MINUTE<900 performs 'Mem1 52; Mem2 60'
10:41:00.301 MQT: stat/geyser/RESULT = {"Mem1":"52; Mem2 60"}
```

i.e. `Mem1` gets set to `52; Mem2 60`, and `Mem2` is unset.

How can I get this to do as I hoped?

**Comment by sfromis** [comment, marked as answer]
When you use multiple commands in a rule, and it is not inside of `if/endif`, you have to use the `Backlog` command, to effectively have multiple commands as one. Without this, your `;` has no function as a command delimiter.

Unrelated, but I'd avoid using Mem variables, unless needed. The point is that each update of those will by default trigger a flash write, and the lifetime of the flash chip depends on how many writes. To be clear, I'm not saying that 5 extra write a day is a problem, when it is not that many after all.

**Reply by garyvdm** [reply]
Thanks, `Backlog` fixed it.

And I've switched from `Mem` to `Var`. After thinking about it, I realized volatile memory is sufficient.
